backend/fms_core/services/utils.py

```diff
@@ -0,0 +1,19 @@
+def fit_string_with_ellipses(string: str, max_length: int, ellipses = "...") -> str:
+    if max_length < 0:
+        raise Exception(f"The max_length '{max_length}' must be a positive integer")
+    if max_length < len(ellipses):
+        raise Exception(f"The max_length '{max_length}' is too short to fit the ellipses '{ellipses}'")
+
+    if len(string) > max_length:
+        if len(string) < len(ellipses):
+            raise Exception(f"The string '{string}' is too short to fit the ellipses '{ellipses}'")
```

Comment:
this case is unreachable since string > max_length and max_length > ellipsis => string > ellipsis.